pkg/core/region.go

```diff
@@ -1516,11 +1516,14 @@ func (r *RegionsInfo) getRegionsByKeys(keys [][]byte) []*RegionInfo {
 	regions := make([]*RegionInfo, 0, len(keys))
 	// Split the keys into multiple batches, and search each batch separately.
 	// This is to avoid the lock contention on the `regionTree`.
+	var results []*RegionInfo
 	for _, batch := range splitKeysIntoBatches(keys) {
 		r.t.RLock()
-		results := r.tree.searchByKeys(batch)
+		results = r.tree.searchByKeys(batch)
+		for _, region := range results {
```

Comment:
What is the difference?

Reply:
`result` is pre-defined to avoid being created multiple times during the batches.

`region` should be retrieved again by ID rather than just using the one from the tree.